indico/modules/designer/models/templates.py

```diff
@@ -130,6 +139,15 @@ def __init__(self, **kwargs):
     def owner(self):
         return self.event_new if self.event_new else self.category
 
+    @property
+    def can_be_deleted(self):
+        from indico.modules.events.registration.models.forms import RegistrationForm
+        active_regforms = RegistrationForm.find_all(RegistrationForm.ticket_template == self,
+                                                    Event.ends_after(now_utc()), _join=Event)
```

Comment:
Please do not use `find()` or `find_all()` with `_join` and similar shortcuts. We pretty much stopped using them many versions ago. I'd actually avoid using the `find` methods in favor of `query` at all - ie do this:

```python
active_regforms = (RegistrationForm.query
                   .filter(RegistrationForm.ticket_template == self,
                           RegistrationForm.event_new.has(Event.ends_after(now_utc())))
                   .all())
```

or this:

```python
active_regforms = (RegistrationForm.query
                   .join(Event)
                   .filter(RegistrationForm.ticket_template == self,
                           Event.ends_after(now_utc()))
                   .all())
```

I think the second one is slighly more performant, but considering that you already have a very small amount of rows due to the filter on ticket_template, I would go for the former since it's more concise.

---

In case you wonder why I'm against using find/find_all/first_first/find_one, even though it was me who added them initially: I thought it'd be nice and when using `indico shell` they are indeed very useful shortcuts. But when writing rael code it's better to stay close to what SQLAlchemy provides out of the box instead of obscure addons - and the various `_...` kwargs of find and friends are /very/ obscure.